torch/fx/experimental/migrate_gradual_types/transform_to_z3.py

```diff
@@ -240,5 +240,84 @@ def transform_all_constraints(traced, counter=0):
 
         return transformed
 
+
+    def transform_all_constraints_trace_time(tracer_root, graph, node, counter=0):
+        """
+        Takes a node and a graph and generates two sets of constraints.
+        One set constraints the node's constraints and another set
+        constraints the negation of the node's constraints
+        Args:
+            tracer_root: the root for getting the module instances
+            graph: the graph so far in the tracing process
+            node: node that represents a conditional
+            counter: variable tracking
+
+        Returns: Two sets of constraints. One with a conjunction with the
+        the conditional constraint and the other with a conjunction with
+        its negation.
+
+        """
+        dimension_dict = {}  # type: ignore[var-annotated]
+
+        generator = ConstraintGenerator(tracer_root, graph)
+        new_constraints, counter = generator.generate_constraints(counter)
+
+        # we know the constraint is a conjunction where the last constraint is about the conditional
+        # so remove the last constraint
+        new_constraints.conjucts = new_constraints.conjucts[:-1]
+
+        condition_constraint, counter = generator.generate_constraints_node(node, counter)
+
+        # transform precision, matching, consistency till obtaining a fixed point
+        old_c = None
+        while old_c != new_constraints:
+            old_c = new_constraints
+            new_constraints, counter = transform_constraint(new_constraints, counter)
+
+        condition_constraint = condition_constraint[0]
+
+        # transform the condition constraint
+        old_c = None
+        while old_c != condition_constraint:
+            old_c = condition_constraint
+            condition_constraint, counter = transform_constraint(condition_constraint, counter)
```

Comment:
As discussed offline, we probably want to hoist the fixpoint portion of this into a function so we don't a) have it repeated and b) expose low-level details

Reply:
done